shell/imports/sandstorm-db/db.js

```diff
@@ -2309,6 +2309,18 @@ SandstormDb.escapeMongoKey = (key) => {
   return key.replace(".", "\uFF0E").replace("$", "\uFF04");
 };
 
+SandstormDb.escapeMongoObject = (obj) => {
+  if (obj && (typeof obj == "object")) {
+    let result = {};
+    for (let key in obj) {
+      result[SandstormDb.escapeMongoKey(key)] = obj[key];
```

Comment:
Oh geez, yes. Great catch.

(Technically not a security bug in the present use case since only admin can access the TLS settings API, but would have been a security bug if we used this somewhere else accessible to normal users.)

Reply:
Looking closer, I think this will also do the wrong thing for arrays: it should traverse them, as `typeof [] === 'object'`, but we'll end up spitting out something that isn't an instance of `Array`, and will be missing things like `length`.

Reply:
Also a great point.